dispute-game: Sanity check disputed claim index

Inphi opened the pull request:
The `IFaultDisputeGame.move` interface has been modified to accept a disputed `Claim` hash. This allows the FDG to sanity check the disputed claim hash against its corresponding index in the `claimData` array.

This improves the UX of the interface to guard against possible reorgs when a challenger attempts to move against a reorg'd claim at an incorrect `claimData` index.

Inphi committed:
dispute-game: Sanity check disputed claim index

clabby reviewed (approved):
Change LGTM. Good to go pending challenger changes.

Also looks like there's some stack too deep issues in the ctb tests to sort.

Inphi committed:
Fix tests

clabby reviewed (changes requested):
Needs CTB snapshots

Inphi committed:
update snapshots

ajsutton commented on packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol:
```diff
@@ -315,13 +315,16 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
     /// @param _claim The claim at the next logical position in the game.
     /// @param _isAttack Whether or not the move is an attack or defense.
-    function move(uint256 _challengeIndex, Claim _claim, bool _isAttack) public payable virtual {
+    function move(Claim _disputed, uint256 _challengeIndex, Claim _claim, bool _isAttack) public payable virtual {
```
The params in the method comment haven't been updated.

ajsutton committed:
op-challenger: Supply parent value when performing moves

Also moves calculation of the required bond into the contract bindings so the returned TxCandidate is ready to send.

Inphi committed:
fix docs

ajsutton committed:
op-challenger: Add backwards compatibility with v1.1.1

Inphi committed:
move param

ajsutton committed:
op-e2e: Update e2e for new FaultDisputeGame ABI
